File: back/src/main/java/com/back/global/security/SecurityUser.java
package com.back.global.security;

import lombok.Getter;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.User;
import org.springframework.security.oauth2.core.user.OAuth2User;

import java.util.Collection;
import java.util.Map;

@Getter
public class SecurityUser extends User implements OAuth2User {
    private final long id;
    private final String name;

    public SecurityUser(
            long id,
            String username,
            String name,
            Collection<? extends GrantedAuthority> authorities
    ) {
        super(username, "", authorities); // 우리의 시나리오(REST API)에서는 이 객체의 비밀번호 필드를 활용할 일이 없다.
        this.id = id;
        this.name = name;
    }

    @Override
    public Map<String, Object> getAttributes() {
        return Map.of();
    }
}

File: back/src/main/kotlin/com/back/global/security/SecurityUser.kt
package com.back.global.security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.User
import org.springframework.security.oauth2.core.user.OAuth2User

class SecurityUser(
    val id: Long,
    username: String,
    private val _name: String,
    authorities: Collection<GrantedAuthority>
) : User(username, "", authorities), OAuth2User {

    override fun getAttributes() = mapOf<String, Any>()
    override fun getName() = _name

}